### ISSUE TITLE
Using https protocol with github causes "fatal: ambiguous argument" error

### ISSUE THREAD

**Issue by crw** [weaveworks/flux#858]
Using github's https protocol scheme results in the following error being displayed in the Weave Cloud interface:

`fatal: ambiguous argument 'flux-dark-sea-36..HEAD': unknown revision or path not in the working tree.`

Presuming this happens with any read-only setup, the expected behavior would be that flux can detect if writes are consistently failing (e.g., only has read-only access to the repository) and display an error message like, "flux requires write access to the repository to correctly function. Please check your configuration."

**Comment by rade** [comment]
> Presuming this happens with any read-only setup

If true that might explain #825.

**Comment by squaremo** [comment]
>> Presuming this happens with any read-only setup
> If true that might explain #825.

The opening comment there mentions using a `git@github.com:...` URL. It's still possible to fail to make your deploy key read/write, of course.

In general it's difficult to detect _why_ git writes are failing, at least while we're exec'ing `git` to get stuff done. (I would love to use something like https://github.com/speedata/gogit but I don't think it supports all the porcelain we need).

**Comment by rade** [comment]
> The opening comment there mentions using a `git@github.com:...` URL. It's still possible to fail to make your deploy key read/write, of course.

My thoughts exactly.

**Comment by squaremo** [comment]
> I would love to use something like https://github.com/speedata/gogit

I meant https://github.com/src-d/go-git, obvs.

**Comment by crw** [comment]
Screenshot of the current error messaging:

![screen shot 2017-11-28 at 3 10 32 pm](https://user-images.githubusercontent.com/57712/33403411-2bc6b250-d515-11e7-8e46-e4e66785037f.png)

**Comment by crw** [comment]
Thinking about this, one thing that is confusing to me from an end-user perspective is that flux is showing an error message that is actually the result of a previous, unreported failure; specifically, not being able to write the tag `flux-dark-sea-36`. If the "couldn't write tag" error was surfaced, it would be a little more obvious that for some reason flux doesn't have write access to the repo; instead it is surfacing an error that should only be possible after we have successfully written a tag (although the error wouldn't occur in that case, obviously).

It is a subtle chain of inference that I am not sure I was even consciously aware of last night, but in any case another reason why that error message is confusing. Another possible solution would be to either link to logs of what happened leading up to that error, or have instructions on how to find those logs attached to the error.

**Comment by squaremo** [comment]
> Thinking about this, one thing that is confusing to me from an end-user perspective is that flux is showing an error message that is actually the result of a previous, unreported failure; specifically, not being able to write the tag flux-dark-sea-36. If the "couldn't write tag" error was surfaced, it would be a little more obvious that for some reason flux doesn't have write access to the repo; instead it is surfacing an error that should only be possible after we have successfully written a tag (although the error wouldn't occur in that case, obviously).

Right, good point. It's not just that it's displaying an error when you're expecting to be done, it's displaying an error that doesn't seem related to what you might expect the problem to be.

I wonder if testing whether the repo is writable before changing state to "ready" would be enough. Probably by attempting to push some innocuous change.
